package YFCC100M;

message ItemBlob{
    required string item_id = 1;

    optional bytes item_data = 2; // byte stream encoding the image data

    required string item_raw_info = 3; // raw information from the link file.

    required string item_display_list  = 4; // a list of interesting info

    enum ItemType {
        IMAGE = 0;
        VIDEO = 1;
        UNKNOWN = 2;
    }

    required ItemType item_type = 5;
}

